app/models/user.rb

```diff
@@ -1,2 +1,5 @@
 class User < ApplicationRecord
+  validates :username, presence: true, uniqueness: true
+  validates :email, presence: true, uniqueness: true
+  validates :password, presence: true, length: { minimum: 8 }
```

Comment:
- [ ] [OPTIONAL] Add some formatting validations to email and password to make sure they are in the expected format, you can use regular expressions for that.